hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java

```diff
@@ -330,6 +336,74 @@ public CleanPlanner(HoodieEngineContext context, HoodieTable<T, I, K, O> hoodieT
     }
     return deletePaths;
   }
+
+  /**
+   * This method finds the files to be cleaned based on the number of hours. If {@code config.getCleanerHoursRetained()} is set to 5,
+   * all the files with commit time earlier than 5 hours will be removed. Also the latest file for any file group is retained.
+   * This policy gives much more flexibility to users for retaining data for running incremental queries as compared to
+   * KEEP_LATEST_COMMITS cleaning policy. The default number of hours is 5.
+   * @param partitionPath partition path to check
+   * @return list of files to clean
+   */
+  private List<CleanFileInfo> getFilesToCleanKeepingLatestHours(String partitionPath) {
```

Comment:
can't we re-use getFilesToCleanKeepingLatestCommits(). all we need to do is to move most of these to a private method and reuse across both. 
for getFilesToCleanKeepingLatestCommits(), you can pass in the config value for N commits to retain. where as for getFilesToCleanKeepingLatestHours, we can determine how many commits can be retained and then pass in the N value.
lets try to re-use code as much as possible.